pandas/tests/test_nanops.py

```diff
@@ -1002,6 +1003,19 @@ def prng(self):
         return np.random.RandomState(1234)
 
 
+def test_add_overflow():
+    # see gh-14068
+    msg = "too (big|large) to convert"
+    with tm.assertRaisesRegexp(OverflowError, msg):
```

Comment:
We don't need to do TDI or DTI specifically, but your point about testing array + scalar (and vice-versa) as well as array + array are well taken.